src/Zendesk.hs

```diff
@@ -141,9 +142,48 @@ runZendeskMain = do
         ShowStatistics -> do
             putTextLn $ "Classifier is going to gather ticket information assigned to: "
                 <> cfgEmail cfg
-            printWarning
             tickets <- runApp (listTickets Assigned) cfg
             printTicketCountMessage tickets (cfgEmail cfg)
+            putTextLn $ "  --Tickets--"
+            -- How many tickets are there
+            putTextLn $ "  Total: " <> show (length tickets)
+            -- How many tickets are open
+            openTickets <- pure $ filterTicketsByStatus tickets "open"
+            putTextLn $ "  Open: " <> show (length openTickets)
+            -- How many tickets are closed
+            closedTickets <- pure $ filterTicketsByStatus tickets "closed"
+            putTextLn $ "  Closed: " <> show (length closedTickets)
+            -- How many open tickets have attachments
+            ticketsWithAttachments <- runApp (filterTicketsWithAttachments closedTickets) cfg
+            putTextLn $ "  Closed with Attachments: " <> show (length ticketsWithAttachments)
+            runApp (mapM_ showTicketAttachments ticketsWithAttachments) cfg
+
+showAttachmentInfo :: Attachment -> App ()
+showAttachmentInfo attachment = do
+    putText "  Attachment: "
+    (putText . show) (aSize attachment)
+    putText " - "
+    putTextLn $ aURL attachment
+
+showCommentAttachments :: Comment -> App ()
+showCommentAttachments comment = do
+    mapM_ showAttachmentInfo (cAttachments comment)
+
+showTicketAttachments :: TicketInfo -> App ()
+showTicketAttachments ticket = do
+    putText "Ticket #: "
+    (putTextLn . show) (ticketId ticket)
+    comments <- getTicketComments (ticketId ticket)
+    mapM_ showCommentAttachments (comments)
+
+filterTicketsWithAttachments :: [TicketInfo] -> App [TicketInfo]
+filterTicketsWithAttachments tickets = do
+    let checkTicketForAttachments :: TicketInfo -> App Bool
```

Comment:
Let me give you an example.
Let's say you have:
```
filterTickets :: TicketInfo -> App Bool
filterTickets ticket = do
    comments <- getTicketComments (ticketId ticket) -- You get comments from a ticket using a monadic context
    pure . null $ comments
```
You start thinking "how can I remove this monadic context and push it out?". Can also be rewritten as:
```
filterTicketByComments :: TicketInfo -> App Bool
filterTicketByComments ticketInfo = do
    comments <- getTicketComments (ticketId ticket) -- You get comments from a ticket using a monadic context
    pure $ filterTickets comments

filterTickets :: [Comment] -> Bool
filterTickets comments = null comments
```
This seems kind of strange but it really isn't. The function `filterTickets` should do what the name says - filter tickets. Not fetch the ticket comments, filter that, and return it.
This buys you composability in the long run.

Reply:
I guess the example here is the fact you can extract the "impure" part out.

Reply:
Ok, reviewing now.

Reply:
isn't filterTickets lacking useful description? To me filterTickets should take something to filter with, not simply test if an array is empty or not. In this case shouldn't it be called something like ticketHasComments or checkTicketForComments?

Reply:
Yes, that's better. I was actually trying to show you a different thing, but you are correct, this is something I wrote down quickly. Let me fix the example.

Reply:
Ok. I will try to port over my work to your recent updates in develop.

Reply:
When I fixed the naming, the example got even more confusing. Let me try with an unrelated and obvious example:
```
printTicket :: TicketInfo -> App ()
printTicket ticketInfo = putStrLn . show $ ticketInfo
```
Can be:
```
showTicket :: TicketInfo -> String
showTicket ticketInfo = show ticketInfo

printTicket :: TicketInfo -> App ()
printTicket ticketInfo = putStrLn $ showTicket ticketInfo
```
See, `showTicket` is now pure. Anyway, hope it help, otherwise ignore my ramblings.

Reply:
So the issue I had is that you need to traverse all the comments of a ticket, then all the attachments of a comment, so to seperate things you'd need [tiicketinfo], [[Comments]], [[[Attachments]]] then try to count them properly (which is prone to error). I'm going to play with the refactoring more and see what's possible.